benchexec/tablegenerator/__init__.py

```diff
@@ -1382,10 +1386,201 @@ def write_head_line(
+                    continue
+                command.set_command_part("stat_type", k if k != "sum" else "")
+                command_list.append(
+                    copy.deepcopy(command).set_command_value(v, **value_data)
+                )
+
+    latex_commands = []
+
+    for run_set, stat_list in zip(run_sets, stats):
+        current_command = LatexCommand(
+            bench_name=run_set.attributes.get("benchmarkname"),
+            runset_name=run_set.attributes.get("displayName"),  # Limiting length
+        )
+        used_column_titles = set()
+        for column, column_stats in zip(run_set.columns, stat_list):
+            column_title = (
+                column.display_title if column.display_title else column.title
+            )
+            if column_title in used_column_titles:
+                logging.warning(
+                    "Detected already used %s! "
+                    "Columns should be unique, please consider changing the name or displayTitle of this column \n"
+                    "Skipping column %s for now" % (column_title, column_title)
+                )
+                continue
+
+            used_column_titles.add(column_title)
+            current_command.set_command_part("column_title", column_title)
+
+            column_statistic_to_latex_command(
+                current_command, column_stats, latex_commands, **column.__dict__
+            )
+
+    out.write(header)
+    out.write("\n".join(command.to_latex_raw() for command in latex_commands))
+
+
+class LatexCommand:
+    """Data holder for latex command."""
+
+    def __init__(self, bench_name="", runset_name=""):
+        self.bench_name = LatexCommand.format_command_part(str(bench_name))
+        self.runset_name = LatexCommand.format_command_part(str(runset_name))
+        self.column_title = ""
+        self.column_category = ""
+        self.column_subcategory = ""
+        self.stat_type = ""
+        self.value = None
+
+    def set_command_part(self, part_name: str, part_value) -> "LatexCommand":
+        """Sets the value of the command part
+
+        Available part names:
+            bench_name, runset_name, column_title, column_category, column_subcategory, stat_type
+
+        Args:
+            part_name: One of the names above
+            part_value: The value to be set for this command part
+
+        Returns:
+            This LatexCommand
+        """
+        self.__dict__[part_name] = LatexCommand.format_command_part(str(part_value))
+        return self
+
+    def set_command_value(self, value, scale_factor, **value_data) -> "LatexCommand":
+        """Sets the value for this command
+
+        Args:
+            value: The new command value
+            scale_factor: Scaling factor for the new value
+            value_data: Remaining unused value_data
+
+        Returns:
+            This LatexCommand
+        """
+        self.value = decimal.Decimal(value)
+        self.value *= scale_factor
```

Comment:
Did you check whether this makes sense? Does the file really contain the same values as the HTML tables? I would say that the statistics values all have the scale factor already applied and now would be wrong. But it needs to be checked.